lib/db_ido_pgsql/idopgsqlconnection.cpp

```diff
@@ -700,9 +700,15 @@ bool IdoPgsqlConnection::FieldToEscapedString(const String& key, const Value& va
 
 		*result = static_cast<long>(dbrefcol);
 	} else if (DbValue::IsTimestamp(value)) {
-		long ts = rawvalue;
+		double ts = rawvalue;
+
+		if (ts < -2e11 || ts > 9e12) {
+			*result = "NULL";
```

Comment:
Why `NULL` instead of [clamping](https://en.cppreference.com/w/cpp/algorithm/clamp) it to the allowed range? Thousands of years in the future still says "absurdly far in the future" better than `NULL`. This would also be more consistent with #10060.

Reply:
Me too (three 😆).

Reply:
Btw, to answer your offline question, the monitoring module shows literally nothing for NULL instead of a timestamp/ETA.